login/profile_blackbox_test.go

```diff
@@ -0,0 +1,164 @@
+// The SetupSuite method will run before the tests in the suite are run.
+// It sets up a database connection for all the tests in this suite without polluting global space.
+func (s *ProfileBlackBoxTest) SetupSuite() {
+
+	var err error
+	s.configuration, err = config.GetConfigurationData()
+	if err != nil {
+		panic(fmt.Errorf("Failed to setup the configuration: %s", err.Error()))
+	}
+
+	keycloakUserProfileService := login.NewKeycloakUserProfileClient()
+	s.profileService = keycloakUserProfileService
+
+}
+
+func (s *ProfileBlackBoxTest) SetupTest() {
+}
+
+func (s *ProfileBlackBoxTest) TearDownTest() {
+}
+
+func (s *ProfileBlackBoxTest) generateAccessToken() (*string, error) {
+
+	var scopes []account.Identity
+	scopes = append(scopes, test.TestIdentity)
+	scopes = append(scopes, test.TestObserverIdentity)
+
+	client := &http.Client{Timeout: 10 * time.Second}
+	r := &goa.RequestData{
+		Request: &http.Request{Host: "api.example.org"},
+	}
+	tokenEndpoint, err := s.configuration.GetKeycloakEndpointToken(r)
+
+	assert.Equal(s.T(), tokenEndpoint, "https://sso.prod-preview.openshift.io/auth/realms/fabric8-test/protocol/openid-connect/token")
+
+	res, err := client.PostForm(tokenEndpoint, url.Values{
+		"client_id":     {s.configuration.GetKeycloakClientID()},
+		"client_secret": {s.configuration.GetKeycloakSecret()},
+		"username":      {s.configuration.GetKeycloakTestUserName()},
+		"password":      {s.configuration.GetKeycloakTestUserSecret()},
+		"grant_type":    {"password"},
+	})
+	if err != nil {
+		return nil, errors.NewInternalError("error when obtaining token " + err.Error())
+	}
+
+	token, err := auth.ReadToken(res)
+	require.Nil(s.T(), err)
+	return token.AccessToken, err
+}
+
+func (s *ProfileBlackBoxTest) TestKeycloakUserProfileUpdate() {
+
+	// UPDATE the user profile
+
+	token, err := s.generateAccessToken() // TODO: Use a simpler way to do this.
+	assert.Nil(s.T(), err)
+
+	// Use the token to update user profile
+
+	testFirstName := "updatedFirstNameAgainNew" + uuid.NewV4().String()
+	testLastName := "updatedLastNameNew" + uuid.NewV4().String()
+	testEmail := "updatedEmail" + uuid.NewV4().String() + "@email.com"
+	testBio := "updatedBioNew" + uuid.NewV4().String()
+	testURL := "updatedURLNew" + uuid.NewV4().String()
+	testImageURL := "updatedBio" + uuid.NewV4().String()
+
+	testKeycloakUserProfileAttributes := &login.KeycloakUserProfileAttributes{
+		login.ImageURLAttributeName: []string{testImageURL},
+		login.BioAttributeName:      []string{testBio},
+		login.URLAttributeName:      []string{testURL},
+	}
+
+	testKeycloakUserProfileData := login.NewKeycloakUserProfile(&testFirstName, &testLastName, &testEmail, testKeycloakUserProfileAttributes)
+
+	r := &goa.RequestData{
+		Request: &http.Request{Host: "api.example.org"},
+	}
+	profileAPIURL, err := s.configuration.GetKeycloakAccountEndpoint(r)
+	assert.Nil(s.T(), err)
+
+	err = s.profileService.Update(testKeycloakUserProfileData, *token, profileAPIURL)
```

Comment:
Done :)

Refactored the tests altogether to avoid repetitive code:
https://github.com/almighty/almighty-core/pull/1035/files#diff-dd1aab321670fa70f6a859b7bcb4400aR72